go/pkg/pgn/config/config.go

```diff
@@ -0,0 +1,157 @@
+	"net"
+
+	"github.com/scionproto/scion/go/lib/addr"
+	"github.com/scionproto/scion/go/lib/config"
+	"github.com/scionproto/scion/go/lib/env"
+	"github.com/scionproto/scion/go/lib/log"
+	"github.com/scionproto/scion/go/lib/serrors"
+)
+
+const (
+	DefaultDb = "./pgn.db"
+)
+
+type Config struct {
+	General  env.General `toml:"general,omitempty"`
+	Features env.Features
+	Logging  log.Config       `toml:"log,omitempty"`
+	Metrics  env.Metrics      `toml:"metrics,omitempty"`
+	Sciond   env.SCIONDClient `toml:"sciond_connection,omitempty"`
+	Pgn      PgnConf          `toml:"pgn,omitempty"`
+}
+
+func (cfg *Config) InitDefaults() {
+	config.InitAll(
+		&cfg.Features,
+		&cfg.Logging,
+		&cfg.Metrics,
+		&cfg.Sciond,
+		&cfg.Pgn,
+	)
+}
+
+func (cfg *Config) Validate() error {
+	return config.ValidateAll(
+		&cfg.Features,
+		&cfg.Logging,
+		&cfg.Metrics,
+		&cfg.Sciond,
+		&cfg.Pgn,
+	)
+}
+
+func (cfg *Config) Sample(dst io.Writer, path config.Path, _ config.CtxMap) {
+	print(path)
+	config.WriteSample(dst, path, config.CtxMap{config.ID: idSample},
+		&cfg.Features,
+		&cfg.Logging,
+		&cfg.Metrics,
+		&cfg.Sciond,
+		&cfg.Pgn,
+	)
+}
+
+var _ config.Config = (*PgnConf)(nil)
+
+type PgnConf struct {
+	// ID of the PGN (required)
+	ID string `toml:"id,omitempty"`
+
+	// DispatcherBypass is the underlay address (e.g. ":30041") to use when bypassing SCION
+	// dispatcher. If the field is empty bypass is not done and SCION dispatcher is used
+	// instead.
+	DispatcherBypass string `toml:"disaptcher_bypass,omitempty"`
+
+	// IP to listen on (required)
+	IP net.IP `toml:"ip,omitempty"`
+
+	// Port to listen on (required)
+	Port uint16 `toml:"port,omitempty"`
+
+	// IA the local IA (required)
+	IA addr.IA `toml:"isd_as,omitempty"`
+
+	//CfgDir directory to read crypto keys from (required)
+	CfgDir string `toml:"cfg_dir,omitempty"`
+
+	//Db to store pgn cfg data (default ./pgn.db will be created or read from)
+	Db string `toml:"db,omitempty"`
+
+	//QUIC address to listen to quic IP:Port (required)
+	QUICAddr string `toml:"quic_addr,omitempty"`
+
+	//CertFile for QUIC socket (required)
+	CertFile string `toml:"cert_file,omitempty"`
+
+	//KeyFile for QUIC socket (required)
+	KeyFile string `toml:"key_file,omitempty"`
+
+	//PLNIA IA of the PLN to contact for PGN lists (required)
+	PLNIA addr.IA `toml:"pln_isd_as,omitempty"`
+}
+
+func (cfg *PgnConf) InitDefaults() {
+	if cfg.Db == "" {
+		cfg.Db = DefaultDb
+	}
+}
+func (cfg *PgnConf) Validate() error {
+	if cfg.ID == "" {
+		return serrors.New("id must be set!")
+	}
+	if cfg.IA.IsZero() {
+		return serrors.New("isd_as must be set")
+	}
+	if cfg.IA.IsWildcard() {
+		return serrors.New("Wildcard isd_as not allowed")
+	}
+	if cfg.IP.IsUnspecified() {
+		return serrors.New("ip must be set")
+	}
+	if cfg.Port == 0 {
+		return serrors.New("Port must be set")
+	}
+	if cfg.CfgDir == "" {
+		return serrors.New("pgn cfg_dir should be set")
+	}
+	if cfg.QUICAddr == "" {
+		return serrors.New("quic addr should be set")
```

Comment:
```suggestion
		return serrors.New("QUIC addr should be set")
```

Use consistently either the struct field name or the toml config key.
Also applies to the PLN PR.